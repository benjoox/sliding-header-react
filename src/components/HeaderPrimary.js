import React, { useRef } from 'react'
import { bool } from 'prop-types'
import { ReactComponent as ForumDots } from './assets/forum-dots.svg'
import { ReactComponent as ForumLogoText } from './assets/forum-logo-text.svg'
import RegisterButton from './RegisterButton'
import SearchBox from './SearchBox'

export default function HeaderPrimary({ collapsed }) {
  const el = useRef(null)

  return (
    <div className="primary" ref={el}>
      <section>
        <div className="vertical-align">
          <ForumDots />
        </div>
        <div
          className={`vertical-align text-logo ${collapsed ? 'collapsed' : ''}`}
        >
          <ForumLogoText />
        </div>
      </section>

      <section>
        <SearchBox collapsed={collapsed} />
        <RegisterButton />
      </section>
    </div>
  )
}

HeaderPrimary.protoType = {
  collapse: bool.isRequired,
}
