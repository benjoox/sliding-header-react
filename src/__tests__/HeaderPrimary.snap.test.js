import React from 'react'
import renderer from 'react-test-renderer'
import HeaderPrimary from '../components/HeaderPrimary'

it('HeaderPrimary  renders correctly', () => {
  const tree = renderer.create(<HeaderPrimary />).toJSON()
  expect(tree).toMatchSnapshot()
})
