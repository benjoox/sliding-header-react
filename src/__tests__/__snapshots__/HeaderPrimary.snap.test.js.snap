// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderPrimary  renders correctly 1`] = `
<div
  className="primary"
>
  <section>
    <div
      className="vertical-align"
    >
      <svg>
        forum-dots.svg
      </svg>
    </div>
    <div
      className="vertical-align text-logo "
    >
      <svg>
        forum-logo-text.svg
      </svg>
    </div>
  </section>
  <section>
    <div
      className="search-box-container"
    >
      <div
        className="search-box "
      >
        <div
          className="search-icon "
        >
          <svg>
            search.svg
          </svg>
        </div>
        <input
          className="search-box-input"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <button
      className="registeration-btn"
    >
      <div>
        <svg>
          profile.svg
        </svg>
      </div>
      <p>
        Register
      </p>
    </button>
  </section>
</div>
`;
